internal/clients/api_client.go

```diff
@@ -82,6 +83,17 @@ func NewApiClientFunc(version string, p *schema.Provider) func(context.Context,
 					}
 					config.Addresses = endpoints
 				}
+				if caFile, ok := esConfig["ca_file"]; ok && caFile.(string) != "" {
+					caCert, err := ioutil.ReadFile(caFile.(string))
+					if err != nil {
+						diags = append(diags, diag.Diagnostic{
+							Severity: diag.Error,
+							Summary:  "Unable to read CA File",
+							Detail:   err.Error(),
+						})
```

Comment:
```suggestion
						})
						return nil, diags
```
It would be great if we immediately return the `diags` here to report the error sooner.